recipes/autoconf/all/test_package/conanfile.py

```diff
@@ -1,47 +1,29 @@
-from conans import AutoToolsBuildEnvironment, ConanFile, tools
-from conan.tools.microsoft import is_msvc
-import contextlib
-import os
-import shutil
+from os import environ
 
-required_conan_version = ">=1.45.0"
+from conan import ConanFile
+from conan.tools.build import can_run
+
+required_conan_version = ">=1.50.0"
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    exports_sources = "configure.ac", "config.h.in", "Makefile.in", "test_package_c.c", "test_package_cpp.cpp",
+    generators = "VirtualBuildEnv"
     test_type = "explicit"
+    win_bash = True
 
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
     def build_requirements(self):
-        self.build_requires(self.tested_reference_str)
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if is_msvc(self):
-            with tools.vcvars(self):
-                with tools.environment_append({"CC": "cl -nologo", "CXX": "cl -nologo",}):
-                    yield
-        else:
-            yield
+        self.tool_requires(self.tested_reference_str)
+        if self._settings_build.os == "Windows" and not environ.get("CONAN_BASH_PATH"):
+            self.tool_requires("msys2/cci.latest")
 
     def build(self):
-        for src in self.exports_sources:
-            shutil.copy(os.path.join(self.source_folder, src), self.build_folder)
-        self.run("{} --verbose".format(os.environ["AUTOCONF"]),
-                 win_bash=tools.os_info.is_windows, run_environment=True)
-        self.run("{} --help".format(os.path.join(self.build_folder, "configure").replace("\\", "/")),
-                 win_bash=tools.os_info.is_windows, run_environment=True)
-        autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        with self._build_context():
-            autotools.configure()
-            autotools.make()
```

Comment:
keep it, with conan v2 flavor please, we must test a build

Reply:
I wasn't aware of the test_v1_package, I will add that

Reply:
Honest question asked out of curiosity; What is the added value of actually compiling something, isn't it enough to just validate of the binaries run? I had this discussion with @memshardedearlier and I can think of arguments for both sides. I'm personally inclined to accept a test which just asserts if the binary runs for an application/tool.

On a separate note, please be aware that we can't actually test the Conan v2 on Windows, since issue https://github.com/conan-io/conan/issues/11975
Prevents running autotools

Reply:
These autotools recipes are important in conan-center, many recipes depend on them for autotools build. This recipe must check that a basic autotools project builds fine.
If you test only a toy usage of this binary, you may miss issues in the recipe.

Reply:
Fair enough. I understand stand reasoning very well. I will revert the V1 functionality tomorrow morning.

Reply:
Not sure how to handle the issue with the V2 though.

I could skip it for Windows and assume if it works for Linux and Mac and runs the simple check on windows that it will work as expected on windows as well.

Is that acceptable?

Reply:
For me it's not acceptable, there are recipes depending on a functional autoconf recipe on Windows. No need to rush, wait msys2 PR. Actually I was planning to submit PR to m4, autoconf, automake and libtool (in this order) afterwards.

Reply:
For us the failing of these chains of recipes (msys2, m4, autotoconf, autotool, gettext, libgettext, cpython, qt) can't be build on Windows with the bash subsystems. These are blocking issues for us at the moment,

It would be ashame if we need to host so many recipes on our own Artifactory server because the CCI versions don't work. It would feel as a big loss for me if the work done on these recipes can't be contributed back to the community and if we both need to maintain parallel recipes for the same dependencies.

Reply:
> For me it's not acceptable, there are recipes depending on a functional autoconf recipe on Windows. No need to rush, wait msys2 PR. Actually I was planning to submit PR to m4, autoconf, automake and libtool (in this order) afterwards.

What would be the main difference between how you would do these recipes and how I made them? I'm willing to put in some extra work to ensure that they're up to the standards of you guys. As I said before it would be a shame if this current effort goes to waste.

Reply:
>  No need to rush, wait msys2 PR.

Ah wait I think some information might have gotten snowed over in all the noise.

The conf issue that I mentioned is still an issue even with your updated msys2 recipe. That is the one I'm using on our Artifactory server

Reply:
@Spacelm I have send you a message on Slack, pls feel free to responsd (or not ;-) ) in your own time